### DISCUSSION TITLE
[Feature Request] `traceCall` events

### DISCUSSION THREAD

**Discussion by hbriese** [zkSync-Community-Hub/zksync-developers#643] [Feature requests]
### Network

zkSync Era

### Provide a brief description of the feature you would like to see implemented.

`traceCall` should return logs in some form or another

## Custom JS tracers
Ideally ZKsync would support [JS tracers like GETH](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-debug#traceconfig)

## Custom tracer
Alternatively, a custom tracer (CallAndEventTracer?) would also trace events

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

This feature would directly benefits developers, particularly but not exclusively wallet developers, by providing them better information about a transaction - thereby indirecty benefiting users.

### Additional Details

Current [CallTracer impl](https://github.com/matter-labs/zksync-era/blob/e44bfffb63634a8989c291e241102ec2b669bfea/core/lib/multivm/src/tracers/call_tracer/vm_1_4_2/mod.rs#L26)

**Comment by bxpana** [comment]
Appreciate the request and will let the team know so that we can try to scope it. If you know more people that would be interested in this, you can get them to upvote so we know that it's something we need to prioritize.

**Comment by roseshahar** [comment]
Ideally we should at least support `withLog` [parameter](https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers#call-tracer) like Geth and Erigon

**Comment by cifer76** [comment]
+1 for this

**Comment by yuval99g** [comment]
+1 for this, very important for web3 developers
